lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python3
+
+'''
+    LCLS Naming Tool
+
+    A tool for checking the form and content (but not meaning) of PV names with
+    respect to the LCLS naming convention.
+
+    Functional component source letters and beampath numbers come from 
+    https://docs.google.com/drawings/d/1Sg7pT2TlKq8PgM7UstmJJBLKeqROQeDWbkD0sDoTDdY/edit
+
+    A valid PV name is in the format FFFFF:GGG:CCC:NN:XXXX, where GGG and NN are
+    optional and can be omitted. The XXXX is omitted for device names.
+
+    All permutations of valid PV and device names according to 
+    LCLS_Photon_Source_and_Systems_Nomenclature.pdf. Names ending in XXXX are 
+    PV names. Without XXXX are device names.
+
+        FFFFF:CCC
+        FFFFF:CCC:NN
+        FFFFF:CCC:XXXX
+        FFFFF:CCC:NN:XXXX
+        FFFFF:GGG:CCC
+        FFFFF:GGG:CCC:NN
+        FFFFF:GGG:CCC:XXXX
+        FFFFF:GGG:CCC:NN:XXXX
+
+    This script only checks the validity of PV and device names.
+'''
+
+import argparse
+import subprocess
+import sys
+import json
+import re
+
+
+fc_dict = {}
+fg_dict = {}
+ccc_dict = {}
+beam_sources = {'K': '', 'L': ''}
+beam_numbers = {'0': '', '1': '', '2': '', '3': '', '4': '', '5': ''}
+
+
+def get_version():
+    result = subprocess.run(
+        ['git', 'describe', '--tags'], 
+        capture_output=True, 
+        text=True
+    )
+    return (str(result.stdout).strip().split('-'))[0]
+
+def display_version():
+    version = get_version()
+
+    parser = argparse.ArgumentParser(description="Current version of LCLS Naming Tool")
+    parser.add_argument('-v', '--version', action='store_true')
+    args = parser.parse_args()
+    
+    if args.version:
+        print(version)
+
+
+def load_taxons():
+    global fc_dict
+    global fg_dict
+    global ccc_dict
+
+    # Read the json files containing all the taxons
+    with open('taxons/functional_component_taxon.json') as fc_file:
+        fc_dict = json.load(fc_file)
+
+    with open('taxons/fungible_element_taxon.json') as fg_file:
+        fg_dict = json.load(fg_file)
+
+    with open('taxons/ccc_taxon.json') as ccc_file:
```

Comment:
These relative filepaths are kind of brittle and rely on us being in a specific directory for everything to work, in a future update we need to make sure these are treated as paths relative to this file instead of relative to the cwd. This will be important if we make this an installed script in the conda environment, which people might run with any arbitrary cwd.

Reply:
I tried a few different ways but couldn't get it to work with the web app (/web/app.py). Will revisit in next PR.